pilot/pkg/bootstrap/server.go

```diff
@@ -1013,16 +1013,16 @@ func getDNSNames(args *PilotArgs, host string) []string {
 
 // createPeerCertVerifier creates a SPIFFE certificate verifier with the current istiod configuration.
 func (s *Server) createPeerCertVerifier(tlsOptions TLSOptions) (*spiffe.PeerCertVerifier, error) {
-	customTLSCertsExists, _, _, _ := hasCustomTLSCerts(tlsOptions)
+	customTLSCertsExists, _, _, caCertPath := hasCustomTLSCerts(tlsOptions)
 	if !customTLSCertsExists && s.CA == nil && features.SpiffeBundleEndpoints == "" && !s.isCADisabled() {
 		// Running locally without configured certs - no TLS mode
 		return nil, nil
 	}
 	peerCertVerifier := spiffe.NewPeerCertVerifier()
 	var rootCertBytes []byte
 	var err error
-	if tlsOptions.CaCertFile != "" {
-		if rootCertBytes, err = os.ReadFile(tlsOptions.CaCertFile); err != nil {
+	if caCertPath != "" {
```

Comment:
seems the TLSOptions struct is really misleading -- you can never read from it directly because you should actually be using hasCustomTLSCerts? Is that accurate?

Reply:
Yes, you're right. I'm thinking if it wouldn't be better if TLSOptions had unexported fields and getters which check default paths. What do you think about such a refactor?

Reply:
unexported on its own doesn't help too much since they are in the same package (at least currently) but the idea seems like it may make things safer

Reply:
Agree, these code is very confusing. Would be better to refactor